### PR TITLE
elasticsearch-operator: updates image to rhel 9

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**
Issue: https://issues.redhat.com/browse/LOG-4065

### CODE REVIEW

File: ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ build_root:
 images:
 - from: base
   inputs:
-    ocp_builder_rhel-8-golang-1.19-openshift-4.12:
+    ocp_builder_rhel-9-golang-1.20-openshift-4.14:
```
Comment (periklis) [inline ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml:66]: This won't work if you do not introduce it in the images section first:
```yaml
  ocp_builder_rhel-9-golang-1.20-openshift-4.14:
    name: builder
    namespace: ocp
tag: rhel-9-golang-1.20-openshift-4.14
```

**Review by periklis — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml (hunk 1 above)
Comment (JoaoBraveCoding) [reply ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml:70]: Yeah I messed up, I reverted everything when I realised I was also updating the ES image and not only the operator image and forgot this

Comment (periklis) [reply ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml:70]: Do we need this one at all then:
```yaml
  ocp_builder_rhel-8-golang-1.19-openshift-4.12:
    name: builder
    namespace: ocp
    tag: rhel-8-golang-1.19-openshift-4.12
```

**Comment by JoaoBraveCoding** [comment]
/pj-rehearse pull-ci-openshift-elasticsearch-operator-master-cluster-logging-operator-e2e

**Comment by JoaoBraveCoding** [comment]
/pj-rehearse pull-ci-openshift-elasticsearch-operator-master-cluster-logging-operator-e2e

**Comment by JoaoBraveCoding** [comment]
/pj-rehearse pull-ci-openshift-elasticsearch-operator-master-e2e-operator
/pj-rehearse pull-ci-openshift-elasticsearch-operator-master-cluster-logging-operator-e2e

**Comment by JoaoBraveCoding** [comment]
/pj-rehearse pull-ci-openshift-elasticsearch-operator-master-cluster-logging-operator-e2e

**Pull request merged**